File: trunk/DuPropioDataScraper/DuProprioLib/Entities/PictureInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using EricUtility;
using System.Web;
using Newtonsoft.Json;

namespace DuProprioLib.Entities
{
    public class PictureInfo
    {
        private int m_Index;
        private string m_Title;
        private string m_Description;
        private string m_ImageUrlSmall;
        private string m_ImageUrlLarge;
        private string m_ImageUrlBig;
        private HouseDetailInfo m_House;

        [JsonIgnore]
        public HouseDetailInfo House
        {
            get { return m_House; }
        }

        public int Index
        {
            get { return m_Index; }
            set { m_Index = value; }
        }

        public string Title
        {
            get { return m_Title; }
            set { m_Title = value; }
        }

        public string Description
        {
            get { return m_Description; }
            set { m_Description = value; }
        }

        public string ImageUrlSmall
        {
            get { return m_ImageUrlSmall; }
            set { m_ImageUrlSmall = value; }
        }

        public string ImageUrlLarge
        {
            get { return m_ImageUrlLarge; }
            set { m_ImageUrlLarge = value; }
        }

        public string ImageUrlBig
        {
            get { return m_ImageUrlBig; }
            set { m_ImageUrlBig = value; }
        }
        public PictureInfo(HouseDetailInfo house, string html)
        {
            m_House = house;
            m_Index = int.Parse(html.Extract("data-index=\"", "\""));
            m_Title = HttpUtility.HtmlDecode(html.Extract("data-title=\"", "\""));
            m_Description = HttpUtility.HtmlDecode(html.Extract("data-desc=\"", "\""));
            m_ImageUrlSmall = html.Extract("data-small=\"", "\"");
            m_ImageUrlLarge = html.Extract("data-large=\"", "\"");
            m_ImageUrlBig = html.Extract("data-big=\"", "\"");
        }
    }
}

File: trunk/DuProprioDataScraperAndroid/src/com/ericmas001/duproprio/entities/PictureInfo.java
package com.ericmas001.duproprio.entities;

import org.json.JSONException;
import org.json.JSONObject;

public class PictureInfo {

//	{
//		"Index" : 0,
//		"Title" : "Photo",
//		"Description" : "",
//		"ImageUrlSmall" : "http://photos.duproprio.com/photo-jumele-a-vendre-levis-quebec-province-small-3036446.jpg",
//		"ImageUrlLarge" : "http://photos.duproprio.com/photo-jumele-a-vendre-levis-quebec-province-large-3036446.jpg",
//		"ImageUrlBig" : "http://photos.duproprio.com/photo-jumele-a-vendre-levis-quebec-province-big-3036446.jpg"
//	}
	
	private int m_Index;
	private String m_Title;
	private String m_Description;
	private String m_ImageUrlSmall;
	private String m_ImageUrlLarge;
	private String m_ImageUrlBig;
	
	public int getIndex() {
		return m_Index;
	}
	public void setIndex(int index) {
		this.m_Index = index;
	}
	public String getTitle() {
		return m_Title;
	}
	public void setTitle(String title) {
		this.m_Title = title;
	}
	public String getDescription() {
		return m_Description;
	}
	public void setDescription(String description) {
		this.m_Description = description;
	}
	public String getImageUrlSmall() {
		return m_ImageUrlSmall;
	}
	public void setImageUrlSmall(String imageUrlSmall) {
		this.m_ImageUrlSmall = imageUrlSmall;
	}
	public String getImageUrlLarge() {
		return m_ImageUrlLarge;
	}
	public void setImageUrlLarge(String imageUrlLarge) {
		this.m_ImageUrlLarge = imageUrlLarge;
	}
	public String getImageUrlBig() {
		return m_ImageUrlBig;
	}
	public void setImageUrlBig(String imageUrlBig) {
		this.m_ImageUrlBig = imageUrlBig;
	}
	
	public PictureInfo(int index, String title, String description,
			String imageUrlSmall, String imageUrlLarge, String imageUrlBig) {
		super();
		this.m_Index = index;
		this.m_Title = title;
		this.m_Description = description;
		this.m_ImageUrlSmall = imageUrlSmall;
		this.m_ImageUrlLarge = imageUrlLarge;
		this.m_ImageUrlBig = imageUrlBig;
	}

	public PictureInfo() {
		super();
	}
	
	public PictureInfo(JSONObject item) throws JSONException {
		super();

		this.m_Index = item.getInt("Index");
		this.m_Title = item.getString("Title");
		this.m_Description = item.getString("Description");
		this.m_ImageUrlSmall = item.getString("ImageUrlSmall");
		this.m_ImageUrlLarge = item.getString("ImageUrlLarge");
		this.m_ImageUrlBig = item.getString("ImageUrlBig");
	}
	
	@Override
	public String toString()
	{
		return m_Index+":"+m_Title;
	}
	
	
}
